client-core/src/service/wallet_service.rs

```diff
@@ -212,17 +208,63 @@ pub fn load_wallet<S: SecureStorage>(
     name: &str,
     enckey: &SecKey,
 ) -> Result<Option<Wallet>> {
-    storage.load_secure(KEYSPACE, name, enckey)
-}
+    let wallet: Option<Wallet> = storage.load_secure(KEYSPACE, name, enckey)?;
+
+    if let Some(value) = wallet {
+        let mut new_wallet = value;
+        // storage -> wallet
+        let info_keyspace = get_info_keyspace(name);
+        new_wallet.view_key = read_pubkey(storage, &info_keyspace, "viewkey")?;
+
+        // pubkey
+        let public_keyspace = get_public_keyspace(name);
+        let private_keyspace = get_private_keyspace(name);
+        let public_keys = storage.keys(&public_keyspace)?;
+        for key in &public_keys {
+            let pubkey = read_pubkey(
+                storage,
+                &public_keyspace,
+                &str::from_utf8(&key).expect("get string in load_wallet"),
+            )?;
+            new_wallet.public_keys.insert(pubkey.clone());
+
+            if let Ok(value) =
+                storage.get_secure(private_keyspace.clone(), pubkey.serialize(), enckey)
+            {
+                if let Some(raw_value) = value {
+                    let privatekey = PrivateKey::deserialize_from(&raw_value)
+                        .expect("get privatekey in load_wallet");
+                    new_wallet.key_pairs.insert(pubkey, privatekey);
+                }
+            }
+        }
```

Comment:
Is there any reason to manualy fetch all these values from different keyspaces instead of encoding at a single place (like it was earlier)?

Reply:
because that make inserting speed O(n), when addresses are 1,0000,000

Reply:
Maybe I'm missing something. But, how does this improve time complexity. This is doing exactly the same thing as earlier (the only different is that now we're storing values in different keyspaces).

Reply:
no, it's very fast
see the benchmark

Reply:
Ok. Maybe I'm not able to see the whole picture. I'll let others comment on this.

Reply:
thanks,
previous code cannot produce 1,000,000 addresses within 5 minutes.
i search many ways to do it.

Reply:
> (the only different is that now we're storing values in different keyspaces).

my guess is that this makes a difference, my understanding is:
- before there was just one key-pair, something like: "wallet-name" -> "wallet-data"
- each address creation, added bytes to wallet-data -- each time this whole thing had to be fetched, decrypted, deserialized etc.
- besides that, it keeps updating this single key -- generally, inserts are faster than updates + may create less storage bloat in internal sled accounting etc.